geocoder_tester/base.py

```diff
@@ -112,14 +114,46 @@ def flat_result(self, result):
             out['distance'] = int(dist.meters)
         return out
 
-
-def search(**params):
-    r = http.get(CONFIG['API_URL'], params=params)
+def search(url, **params):
+    r = http.get(url, params=params,
+                 headers={'user-agent': 'geocode-tester/0.0.1'})
     if not r.status_code == 200:
         raise HttpSearchException(error="Non 200 response")
     return r.json()
 
 
+def search_params_default(query, limit, lang, center):
```

Comment:
I think it would make sense here to use classes and inheritance instead.

Something like:

```
class Search:

    def __call__(self, **params):
        …

    def get_url(self):
        …

    def make_params(self, params):
        …

class Photon(Search):

    def get_url(self):
        return super().get_url() + "/api"


class Nominatim(Search):

    def make_params(self, params):
        # do something specific

```

And then:

```
API_TYPES = {
    "default": Search,
    "nominatim": Nominatim,
    "photon": Photon,
}

search = API_TYPES.get(CONFIG['API_TYPE'])
search(**params)
```

What do you think ? :)

Reply:
Yes, you are right. The `_call_` function should be the same everywhere, so I left that, otherwise adapted as you suggested.